internal/types/document_validation.go

```diff
@@ -67,7 +67,6 @@ func (d *Document) ValidateData() error {
 	duplicateChecker := make(map[string]struct{}, len(keys))
 	var idPresent bool
 
-	// TODO: make sure that `_id` is the first item in the map
```

Comment:
👍

Reply:
TBH, I don't understand why we did not do that there and instead extracted an exported function for something that should be an internal detail

Reply:
I had the same comment initially, but at some point I understand the idea of splitting the stage where we "normalize" data and where we "validate" data.

Reply:
That might be a good idea, but that doesn't explain why we do that in handlers, of all things. That's too late for that. I still think that should be an internal detail of `type` package

Reply:
@AlekSi do you have a more exact suggestion where this could be moved? To which place/stage?

Reply:
Should I move `MoveIDToTheFirstIndex` inside `ValidateData`?